broker/src/main/java/io/zeebe/broker/system/partitions/impl/PartitionTransitionImpl.java

```diff
@@ -41,27 +41,48 @@ public PartitionTransitionImpl(
   @Override
   public ActorFuture<Void> toFollower() {
     final CompletableActorFuture<Void> future = new CompletableActorFuture<>();
-    currentTransition.onComplete((nothing, err) -> transition(future, followerSteps));
+    enqueueTransition(future, followerSteps);
     return future;
   }
 
   @Override
   public ActorFuture<Void> toLeader() {
     final CompletableActorFuture<Void> future = new CompletableActorFuture<>();
-    currentTransition.onComplete((nothing, err) -> transition(future, leaderSteps));
+    enqueueTransition(future, leaderSteps);
     return future;
   }
 
   @Override
   public ActorFuture<Void> toInactive() {
     final CompletableActorFuture<Void> future = new CompletableActorFuture<>();
```

Comment:
You can move the creation of this future into the enqueueTransition method and simply return it from there. That would reduce each of these methods to a single return statement.